libmamba/src/specs/version_spec.cpp

```diff
@@ -65,6 +68,91 @@ namespace mamba::specs
         return lhs.level == rhs.level;
     }
 
+    namespace
+    {
+        auto version_match_glob(const CommonVersion& candidate, const CommonVersion& pattern) -> bool
+        {
+            auto cand_it = candidate.cbegin();
+            const auto cand_last = candidate.cend();
+            auto pat_it = pattern.cbegin();
+            const auto pat_last = pattern.cend();
+            auto parts_required = std::size_t(0);
+            auto parts_available = std::size_t(0);
+
+            constexpr auto is_failed = [](auto req, auto avail) -> bool
+            { return (req > avail) || (req == 0 && avail > 0); };
+            constexpr auto is_glob_part = [](const auto& p) -> bool
+            { return p == VERSION_GLOB_SEGMENT; };
+            constexpr auto distance = [](auto i1, auto i2)
+            {
+                assert(i1 <= i2);
+                return static_cast<std::size_t>(std::distance(i1, i2));
+            };
+
+            while (pat_it != pat_last)
+            {
+                // We move forward in the pattern counting all the contiguous glob parts.
+                const auto pat_sub_first = std::find_if_not(pat_it, pat_last, is_glob_part);
+                parts_required += distance(pat_it, pat_sub_first);
+                pat_it = pat_sub_first;
+
+                // No more explicit subpatterns (i.e. not globs) to match
+                if (pat_it == pat_last)
+                {
+                    break;
+                }
+
+                // Find the end of the sub pattern, i.e. to the start of the next glob.
+                // This is required to avoid greedily matching on the first similar character.
+                const auto pat_sub_last = std::find_if(pat_sub_first, pat_last, is_glob_part);
+
+                // At this cand we have a required pattern.
```

Comment:
What do you mean by `cand`? Candidate?

Reply:
candidate, yes. This was a brute find-replace